### PR TITLE
feat: Enable to encrypt and decrypt a string by ansible-vault

### PR DESCRIPTION
**Pull request by hsong-rh**
https://issues.redhat.com/browse/AAP-19426

### CODE REVIEW

**Comment by hsong-rh** [comment]
@Alex-Izquierdo `ansible-vault` is found, even though I didn't change CI workflow. Is this because it's added in Dockerfile image?

**Comment by Alex-Izquierdo** [comment]
@hsong-rh ansible-core stills be in the project dependencies. It is going to be removed in https://github.com/ansible/eda-server/pull/679

**Comment by Alex-Izquierdo** [comment]
Anyway, even we remove i, it could work on CI, because the worker could have installed ansible (as well as it has a lot of common applications for CI) But even in that case, It should be present in the workflow, just to make it explicit.

File: src/aap_eda/api/vault.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+class AnsibleVaultEncryptionFailed(Exception):
+    pass
+
+
+class AnsibleVaultDecryptionFailed(Exception):
+    pass
+
+
+def encrypt_string(password: str, plaintext: str, vault_id: str) -> str:
+    if not shutil.which("ansible-vault"):
```
Comment (Alex-Izquierdo) [inline src/aap_eda/api/vault.py:33]: I think we want to raise the error at startup as we do with git: 
https://github.com/ansible/eda-server/blob/main/src/aap_eda/services/project/git.py#L47

The concern of @benthomasson is to not use ansible as library, but even through the CLI, the presence of ansible-vault is still a requirement of the application.

File: src/aap_eda/api/vault.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    child.sendcontrol("D")
+    i = child.expect(["Decryption successful", "ERROR"])
+    if i == 0:
+        child.readline()
+        data = child.readline()
+        decrypted_text = ""
+        while data:
+            decrypted_text += data.decode()
+            data = child.readline()
+        return decrypted_text.strip()
```
Comment (Alex-Izquierdo) [inline src/aap_eda/api/vault.py:79]:
```suggestion
        decrypted_text = "".join(line.decode() for line in child)
        return decrypted_text.strip()
```

File: src/aap_eda/api/vault.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    i = child.expect(["Encryption successful", "ERROR"])
+    if i == 0:
+        child.readline()
+        data = child.readline()
+        encrypted_text = ""
+        while data:
+            enc_str = data.decode()
+            if not enc_str.startswith("!vault"):
+                encrypted_text += enc_str.lstrip()
+            data = child.readline()
```
Comment (Alex-Izquierdo) [inline src/aap_eda/api/vault.py:54]:
```suggestion
        encrypted_text = "".join(line.decode() for line in child if not line.startswith("!vault"))
```

File: src/aap_eda/api/vault.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import shutil
+import tempfile
+
+import pexpect
+
+
+class AnsibleVaultNotFound(Exception):
+    pass
```
Comment (Alex-Izquierdo) [inline src/aap_eda/api/vault.py:21]: Should the exception have a clear message like with do with git:
https://github.com/ansible/eda-server/blob/main/src/aap_eda/services/project/git.py#L49

**Review by Alex-Izquierdo — DISMISSED** [review]

**Review by bzwei — APPROVED** [review]

**Review by mkanoor — APPROVED** [review]

**Pull request merged**